### PR TITLE
Update for not displaying blue notice in GA4 setup.

### PR DESCRIPTION
**Pull request by Hazlitte**
## Summary

Update for not displaying blue notice in GA4 setup.

Addresses issue #3614

## Checklist

- [X] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [X] I have added a QA Brief on the issue linked above.
- [X] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js [hunk 1]
```diff
@@ -77,9 +78,9 @@ export default function SetupFormGA4() {
 				<GA4PropertySelect />
 			</div>
 
-			<GA4PropertyNotice
+			{ ga4PropertyID && ( ga4PropertyID !== PROPERTY_CREATE ) && <GA4PropertyNotice
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4.js:81]: See https://github.com/google/site-kit-wp/issues/3614#issuecomment-875856859, only the first condition here must be kept. Only if no property selection has been made (setting up a new property _is_ a selection), the notice must not be shown.

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js [hunk 2]
```diff
@@ -100,7 +101,7 @@ export default function SetupFormGA4Transitional() {
 				</div>
 			) }
 
-			<GA4PropertyNotice notice={ notice }>
+			{ ( propertyID || ga4PropertyID ) && ( ga4PropertyID !== PROPERTY_CREATE ) && <GA4PropertyNotice notice={ notice }>
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:104]: See above.

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js [hunk 3]
```diff
@@ -51,6 +51,7 @@ export default function SetupFormGA4Transitional() {
 	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
 	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
 
+	const ga4PropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID() );
 	const ga4ExistingTag = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getExistingTag() );
 	const ga4MeasurementID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getMeasurementID() );
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:56]: Maybe for readability, we could put the condition here into a `shouldShowAssociatedPropertyNotice` variable, similar to how it is below in `assets/js/modules/analytics/components/setup/SetupFormUA.js`.

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ const { useSelect, useDispatch } = Data;
 export default function SetupFormGA4() {
 	const accounts = useSelect( ( select ) => select( STORE_NAME ).getAccounts() ) || [];
 
+	const ga4PropertyID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID() );
 	const ga4HasExistingTag = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).hasExistingTag() );
 	const ga4ExistingTag = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getExistingTag() );
 	const ga4MeasurementID = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getMeasurementID() );
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4.js:48]: See my other comment, might be helpful to put the overall condition for the notice into a variable.

**Review by felixarntz — CHANGES_REQUESTED** [review]

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js (hunk 1 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4.js:81]: 👍

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js (hunk 2 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:104]: 👍

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js (hunk 3 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:56]: 👍

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js (hunk 4 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4.js:48]: 👍

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js [hunk 5]
```diff
@@ -50,6 +50,8 @@ export default function SetupFormGA4() {
 	const { setValues } = useDispatch( CORE_FORMS );
 	const { setUseSnippet } = useDispatch( MODULES_ANALYTICS_4 );
 
+	const shouldShowAssociatedPropertyNotice = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getPropertyID() );
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4.js:53]: We may want to actually assign this to a `ga4Property` variable first and then include the same conditions as below `accountID && accountID !== ACCOUNT_CREATE && ga4PropertyID`. These should be consistent between all three scenarios:

1. UA scenario: require account not empty and not create, and require UA property not empty
2. GA4 scenario: require account not empty and not create, and require GA4 property not empty
3. GA4-transitional scenario: require account not empty and not create, and require UA property _or_ GA4 property not empty

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js [hunk 6]
```diff
@@ -51,12 +51,15 @@ export default function SetupFormGA4Transitional() {
...
 
 	const { setUseSnippet: uaSetUseSnippet } = useDispatch( STORE_NAME );
 	const { setUseSnippet: ga4SetUseSnippet } = useDispatch( MODULES_ANALYTICS_4 );
 
+	const shouldShowAssociatedPropertyNotice = propertyID || ga4PropertyID;
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:61]: See above, let's make this consistent with the other two scenarios: `accountID && accountID !== ACCOUNT_CREATE && ( propertyID || ga4PropertyID )`

**Review by felixarntz — CHANGES_REQUESTED** [review]
@Hazlitte Two small extra changes here, but then this should be good to go!

File: assets/js/modules/analytics/components/setup/SetupFormGA4.js (hunk 5 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4.js:53]: 👍

File: assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js (hunk 6 above)
Comment (Hazlitte) [reply assets/js/modules/analytics/components/setup/SetupFormGA4Transitional.js:61]: 👍

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**